### PR TITLE
Feature/improve-sonarcloud

### PR DESCRIPTION
**Pull request by kema91**

### CODE REVIEW

File: src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+import de.retest.ui.descriptors.Attributes;
+import de.retest.ui.descriptors.Element;
+import de.retest.ui.descriptors.IdentifyingAttributes;
+import de.retest.ui.descriptors.RootElement;
+
+class ByBestMatchToRetestIdTest {
+
+	@Test
+	void constructor_should_throw_new_illegal_argument_exception() {
```
Comment (oezelenes) [inline src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java]: Method name does not need the operator `new` for a exception.

File: src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		assertThatThrownBy( () -> new ByBestMatchToRetestId( "" ) ) //
+				.isInstanceOf( IllegalArgumentException.class ) //
+				.hasMessage( "retestId must not be empty." );
+	}
+
+	@Test
+	void find_elements_should_throw_new_no_element_with_retest_id_found_exception() {
```
Comment (oezelenes) [inline src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java]: Method name does not need the operator `new` for a exception.

File: src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java (hunk 1 above)
Comment (beatngu13) [inline src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java]: Test names should follow the [should pattern](https://stackoverflow.com/questions/155436/unit-test-naming-best-practices/7398606#7398606). That is, the test name should suggest what happens und which circumstances, e.g., `ctor_should_throw_new_illegal_argument_exception_when_retest_id_is_empty`.

File: src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+		final RootElement lastExpected =
+				new RootElement( "bar", identifyingAtt, attributes, null, Arrays.asList( element ), null, 0, null );
+		final RootElement lastActual =
+				new RootElement( "he", identifyingAtt, attributes, null, Arrays.asList( element ), null, 0, null );
+		final ByBestMatchToRetestId cut = new ByBestMatchToRetestId( "foo" );
+		assertThat( cut.findElement( lastExpected, lastActual ) ).isEqualTo( element );
+	}
+
+	@Test
+	void hash_code_should_return_an_expected_value() {
```
Comment (beatngu13) [inline src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java]: Does this test make sense?

File: src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+		final ByBestMatchToRetestId cut = new ByBestMatchToRetestId( "foo" );
+		final int expected = 101574;
```
Comment (oezelenes) [inline src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java]: You can do something like: 
`String foo = "foo";`
`final int expected = foo.hashCode();`

File: src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+		assertThat( cut.hashCode() ).isEqualTo( expected );
+	}
+
+	@Test
+	void equals_should_return_false() {
```
Comment (oezelenes) [inline src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java]: `equal_should_return_false_when_null` or something like that?

File: src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+		final ByBestMatchToRetestId cut = new ByBestMatchToRetestId( "foo" );
+		assertThat( cut.equals( null ) ).isFalse();
+	}
+
+	@Test
+	void equals_should_return_true() {
```
Comment (oezelenes) [inline src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java]: `equals_should_return_false_when_same_object` or something like that?

File: src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+		final ByBestMatchToRetestId cut = new ByBestMatchToRetestId( "foo" );
+		assertThat( cut.equals( cut ) ).isTrue();
+	}
+
+	@Test
+	void equals_should_return_false_if_the_class_are_different() {
```
Comment (oezelenes) [inline src/test/java/de/retest/web/selenium/ByBestMatchToRetestIdTest.java]: classes instead of class.

**Pull request merged**